src/tera.rs

```diff
@@ -461,17 +462,17 @@ impl Tera {
     /// If a filter with that name already exists, it will be overwritten
     ///
     /// ```rust,ignore
-    /// tera.register_filter("upper", string::upper);
+    /// tera.register_filter("upper", &make_filter_fn(string::upper);
     /// ```
-    pub fn register_filter(&mut self, name: &str, filter: FilterFn) {
-        self.filters.insert(name.to_string(), filter);
+    pub fn register_filter(&mut self, name: &str, filter: &Rc<dyn FilterFn>) {
```

Comment:
I haven't used `impl Trait` yet but is not possible to have `pub fn register_filter(&mut self, name: &str, filter: &impl FilterFn)` and do the `Rc` ourselves?

Reply:
I don't think so. To make an Rc register-functions should get ownership of a function.

Reply:
Also, Rc is a part of public API, because of public fields of struct `Tera`.

Reply:
Ah yes I forgot it can now have a context and isn't just a function